internal/fs/vss_windows.go

```diff
@@ -738,25 +749,14 @@ func NewVssSnapshot(
 
 	timeoutInMillis := uint32(timeoutInSeconds * 1000)
 
-	oleIUnknown, result, err := initializeVssCOMInterface()
+	oleIUnknown, _, err := initializeVssCOMInterface()
 	if oleIUnknown != nil {
 		defer oleIUnknown.Release()
 	}
```

Comment:
And now you can call `defer oleIUnknown.Release()` unconditionally after next `if err != nil` statement.

Reply:
I prefer to keep it conditional here so it gets cleaned up even if I messed up somewhere else.

Reply:
I totally agree with @greatroar here and it's common for other languages too: errors and values mutually exclusive. In languages with exception support you simple can't get value if exception occurred. Go goes the same way, but you have to do everything yourself.